compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -467,7 +467,8 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
             }
             (VarValue::Empty(a_ui), VarValue::Value(b)) => {
                 match *b {
-                    ReLateBound(..) | ReErased | ReError(_) => {
+                    ReError(_) => false,
```

Comment:
Spacing might make this clearer. Maybe this could use a comment about how this is always on an error path, so it doesn't really matter if it's shorter or longer than an empty region.

```suggestion
                    ReError(_) => false,

```